IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/Browser.cs

```diff
@@ -130,13 +130,37 @@ public bool WaitUntil(Func<bool> func, int secondsToWait = 15)
         /// </summary>
         public void TakeScreenshot()
         {
-            string fullPath = Path.Combine(Directory.GetCurrentDirectory(), "screenshot", $"{((RemoteWebDriver)this.Driver).Capabilities.BrowserName}_{DateTime.Now:yyyy.MM.dd_hh.mm.ss}.png");
-            Console.WriteLine($"Saving screenshot to location: {fullPath}");
-            Directory.CreateDirectory(Path.Combine(Directory.GetCurrentDirectory(), "screenshot"));
+            string path = Path.Combine(Directory.GetCurrentDirectory(), "screenshot");
+            Directory.CreateDirectory(path);
+            TakeScreenshot(path, $"{((RemoteWebDriver)this.Driver).Capabilities.BrowserName}_{DateTime.Now:yyyy.MM.dd_hh.mm.ss}.png");
+        }
+
+        /// <summary>
+        /// Take a screenshot of the browser and save it to the passed in fully qualified path.
+        /// Will not throw if the path does not exist.
+        /// </summary>
+        /// <param name="path">The fully qualified path to save the screenshot to</param>
+        /// <param name="fileName">The name for the screenshot file</param>
+        public void TakeScreenshot(string path, string fileName)
+        {
+            if (string.IsNullOrWhiteSpace(path) || string.IsNullOrWhiteSpace(fileName))
+            {
+                Console.WriteLine($"Skipping TakeScreenshot. No path or filename handed into method.");
+                return;
+            }
+
+            if (!Directory.Exists(path))
```

Comment:
I was thinking about that but wasn't sure how much the framework should muck about "behind the scenes". I guess I don't have strong feelings either way, so can add it if there are no other concerns to to speak of.

Reply:
In general I like moving it up to the caller only because it keeps the responsibility of this method down to one thing.

Reply:
Makes sense. Leaving as is. Open to suggestions for handling the error condition better (I feel like outright throwing isn't appropriate as that would fail the test in question, but am open to that if there's no better way to log the info out.)